lib/signaling/v2/peerconnection.js

```diff
@@ -634,6 +634,25 @@ function getConfiguration(configuration) {
 }
 
 /**
+ * Get the m= sections of a particular kind and direction from an sdp.
+ * @param {string} sdp -  sdp string
+ * @param {string} [kind] - Pattern for matching kind
+ * @param {string} [direction] - Pattern for matching direction
+ * @returns {Array<string>} mediaSections
+ */
+function getMediaSections(sdp, kind, direction) {
+  kind = kind || '.*';
+  direction = direction || '.*';
+  return sdp.split('\r\nm=').slice(1).map(function(mediaSection) {
+    return 'm=' + mediaSection;
+  }).filter(function(mediaSection) {
+    var kindPattern = new RegExp('m=' + kind, 'gm');
+    var directionPattern = new RegExp('a=' + direction, 'gm');
```

Comment:
@ryan-rowland No, because we have to filter out only those `m=` sections that match the `kind` and `direction`. Doing it outside the loop just tells you whether the SDP has at least one `m=` section of the given `kind` and `direction`.